File: backend/app/schemas/prediction.py
from pydantic import BaseModel


CLASSES = ["akiec", "bcc", "bkl", "df", "mel", "nv", "vasc"]
CLASS_NAMES = {
    "akiec": "Actinic Keratosis",
    "bcc":   "Basal Cell Carcinoma",
    "bkl":   "Benign Keratosis",
    "df":    "Dermatofibroma",
    "mel":   "Melanoma",
    "nv":    "Melanocytic Nevus",
    "vasc":  "Vascular Lesion",
}


class BoundingBox(BaseModel):
    x1: int
    y1: int
    x2: int
    y2: int


class Detection(BaseModel):
    class_id: int
    class_code: str
    class_name: str
    confidence: float
    bbox: BoundingBox


class PredictionResponse(BaseModel):
    detections: list[Detection]
    image_width: int
    image_height: int

File: mobile/lib/models/prediction.dart
class BoundingBox {
  final int x1, y1, x2, y2;
  const BoundingBox({
    required this.x1,
    required this.y1,
    required this.x2,
    required this.y2,
  });

  factory BoundingBox.fromJson(Map<String, dynamic> json) => BoundingBox(
        x1: json['x1'],
        y1: json['y1'],
        x2: json['x2'],
        y2: json['y2'],
      );
}

class Detection {
  final int classId;
  final String classCode;
  final String className;
  final double confidence;
  final BoundingBox bbox;

  const Detection({
    required this.classId,
    required this.classCode,
    required this.className,
    required this.confidence,
    required this.bbox,
  });

  factory Detection.fromJson(Map<String, dynamic> json) => Detection(
        classId: json['class_id'],
        classCode: json['class_code'],
        className: json['class_name'],
        confidence: (json['confidence'] as num).toDouble(),
        bbox: BoundingBox.fromJson(json['bbox']),
      );
}

class PredictionResponse {
  final List<Detection> detections;
  final int imageWidth;
  final int imageHeight;

  const PredictionResponse({
    required this.detections,
    required this.imageWidth,
    required this.imageHeight,
  });

  factory PredictionResponse.fromJson(Map<String, dynamic> json) =>
      PredictionResponse(
        detections: (json['detections'] as List)
            .map((d) => Detection.fromJson(d))
            .toList(),
        imageWidth: json['image_width'],
        imageHeight: json['image_height'],
      );
}
